src/acceptance/app/lead_times_test.go

```diff
@@ -0,0 +1,60 @@
+package app_test
+
+import (
+	. "acceptance/helpers"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Autoscaler lead times for scaling", func() {
+	var (
+		policy string
+		err    error
+	)
+	BeforeEach(func() {
+		policy = GenerateDynamicScaleOutAndInPolicy(1, 2, "test_metric", 500, 500)
+		appName = CreateTestApp(cfg, "labeled-go_app", 1)
+		appGUID, err = GetAppGuid(cfg, appName)
+		Expect(err).NotTo(HaveOccurred())
+		instanceName = CreatePolicy(cfg, appName, appGUID, policy)
+		StartApp(appName, cfg.CfPushTimeoutDuration())
+	})
+	AfterEach(AppAfterEach)
+
+	When("breach_duration_secs and cool_down_secs are set", func() {
```

Comment:
I ran this test here 20 times just to be sure that it is not flak(e)y: https://github.com/cloudfoundry/app-autoscaler-release/pull/2962/commits/61183d4f0250cb653a61a6b43072b0d23c1a5c8c